android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/favourites/FavoritesScreen.kt

```diff
@@ -79,21 +75,19 @@ fun FavoritesScreen(
                     imageUrl = movie.imageUrl,
                     isFavorite = movie.isFavorite
                 ),
-                onClick = { onNavigateToMovieDetails(movie) }
-            ) { }
+                onClick = { onNavigateToMovieDetails(movie) },
+                onFavouriteButtonClick = onFavoriteButtonClick
+            )
         }
     }
 }
 
 @Preview(showBackground = true)
 @Composable
 fun FavouritesScreenPreview() {
-    val favoritesMapper = FavoritesMapperImpl()
-
     MovieAppTheme {
-        FavoritesScreen(
-            favoritesViewState = favoritesMapper.toFavoritesState(getMoviesList()),
-            onNavigateToMovieDetails = { }
-        )
+        FavoritesRoute(
```

Comment:
Routes should be used for navigation and state hoisting. 
Use pure composable functions for preview. In this case: FavoritesScreen.